astropy/table/tests/test_table.py

```diff
@@ -3178,6 +3178,12 @@ def test_rows_with_mixins():
     t.group_by("obs")
 
 
+def test_group_by_empty_table():
+    # see https://github.com/astropy/astropy/issues/11884
+    t = Table(names=["a", "b"])
+    t.group_by("a")
```

Comment:
Do we want to check the state of `t` after this or it doesn't matter?

Reply:
I think we should, but I don't have a clear idea what to check for.

Reply:
OK hopefully @taldcroft can advise.

Reply:
Let's remind @taldcroft about this :)